### DISCUSSION TITLE
JSON Validation with JSONEditor

### DISCUSSION THREAD

**Discussion by jm-amt** [zauberzeug/nicegui#4003] [Ideas / Feature Requests]
### Question

Is it possible to use the [validator](https://github.com/josdejong/svelte-jsoneditor?tab=readme-ov-file#validator) functionality in the JSONEditor through nicegui? 
I have jsonschema that I want to pass through and have the JSONEditor validate.

**Comment by falkoschindler** [comment]
Hi @jm-amt,

I don't think this is currently possible. But I'm not too familiar with the JSONEditor and Ajv's validator. Maybe there's a way to instantiate the validator via `ui.add_head_html` and somehow add it to the `ui.json_editor` later.

Anyway, I'll convert this discussion into a feature request. Maybe someone from the community can help out to find a solution or at least a workaround.

**Reply by jm-amt** [reply]
Would you accept a PR adding a `json_editor.add_validation(schema)` method? 
Looks like adding this to json_editor.js would work
```js 
add_validation(schema) {
  if (this.editor) {
    const validator = createAjvValidator({ schema: schema, schemaDefinitions: {}, ajvOptions: {} });
    this.editor.updateProps({ ...this.props, validator: validator });
  }
},
```

**Reply by falkoschindler** [reply]
@jm-amt Sure, a pull request is a great way to further discuss an idea. So far I don't see where `createAjvValidator` is coming from and how to use `add_validation`.

**Reply by jm-amt** [reply]
I created a PR with the needed changes. `createAjvValidator` is just imported along side `JSONEditor` from the json_editor module
#4012
